src/main/java/de/taimos/pipeline/aws/WithAWSStep.java

```diff
@@ -123,47 +127,65 @@ public String getExternalId() {
 	public void setExternalId(String externalId) {
 		this.externalId = externalId;
 	}
-	
+
 	@Extension
 	public static class DescriptorImpl extends AbstractStepDescriptorImpl {
-		
+
 		public DescriptorImpl() {
 			super(Execution.class);
 		}
-		
+
 		@Override
 		public String getFunctionName() {
 			return "withAWS";
 		}
-		
+
 		@Override
 		public String getDisplayName() {
 			return "set AWS settings for nested block";
 		}
-		
+
 		@Override
 		public boolean takesImplicitBlockArgument() {
 			return true;
 		}
+
+		public ListBoxModel doFillCredentialsItems(@AncestorInPath Item context) {
+
+			if (context == null || !context.hasPermission(Item.CONFIGURE)) {
+				return new ListBoxModel();
+			}
+
+			return new StandardListBoxModel()
+					.includeEmptyValue()
+					.includeMatchingAs(
+					        context instanceof Queue.Task
+                                    ? Tasks.getAuthenticationOf((Queue.Task) context)
+                                    : ACL.SYSTEM,
+						    context,
+                            StandardUsernamePasswordCredentials.class,
```

Comment:
The only cons I can think of is that `AmazonWebServicesCredentials` carries a redundant information (the role). The `withAWS` step has a parameter for this already. So it could be confusing or it could be integrated accordingly with one taking precedence (the `role` specified in the `withAWS` step) over the other (the one specified in the AWS credentials).

> Support both types would be nice but I'm not sure it is easy.

As far as I can tell it would require to find credentials of type `StandardCredentials` and then do a conditional on the type of implementation. Could be part of a different issue maybe ?

Reply:
> Could be part of a different issue maybe ?

It would definitively be a different Jira issue. I'm not sure that the Credentials API easily supports it at the moment.